Packages/vcs/Lib/VTKAnimate.py

```diff
@@ -7,15 +7,92 @@
 import hashlib
 import os
 import glob
+import vtk
+
+
+def update_input(controller, update=True):
+    ## Ok let's loop through the arrays and figure out the slice needed and update
+    for i,info in enumerate(controller.vcs_self.animate_info):
+      disp,slabs = info
+      slab = slabs[0]
+      if slab is None:
+        continue # nothing to do
+      #Ok we have a slab, let's figure which slice it is
+      args=[]
+      Ntot=1
+      for a in slab.getAxisList()[:-controller._number_of_dims_used_for_plot][::-1]:
+         n=controller.frame_num/Ntot % len(a)
+         Ntot*=len(a)
+         args.append(slice(n,n+1))
+      args=args[::-1]
+      if slabs[1] is None:
+          controller.vcs_self.backend.update_input(disp.backend,slab(*args),update=update)
+      else:
+          controller.vcs_self.backend.update_input(disp.backend,slab(*args),slabs[1](*args),update=update)
 
 class VTKAnimationCreate(animate_helper.StoppableThread):
   def __init__(self, controller):
     animate_helper.StoppableThread.__init__(self)
+
     self.controller = controller
 
-  def run(self):
-    self.controller.animation_created = True
+    # this all happens in the init function because in interaction mode, run basically never happens
+    self.hidden_window = vtk.vtkRenderWindow()
+    self.hidden_window.SetOffScreenRendering(1)
     self.controller._unique_prefix=hashlib.sha1(time.asctime()+str(random.randint(0,10000))).hexdigest()
+    self.bg_ren = vtk.vtkRenderer()
+    self.hidden_window.SetNumberOfLayers(2)
+    self.bg_ren.SetLayer(0)
+    self.hidden_window.AddRenderer(self.bg_ren)
+    self.controller.animation_created = True
+
+  def run(self):
+    pass
+
+  def get_frame_name(self, frame_num):
+    png_name = os.path.join(os.environ["HOME"],".uvcdat",self.controller._unique_prefix,"anim_%i.png" % frame_num)
+    if not os.path.exists(os.path.dirname(png_name)):
+        os.makedirs(os.path.dirname(png_name))
+    return png_name
+
+  def get_frame(self, frame_num):
+    png_name = self.get_frame_name(frame_num)
+
+    if not os.path.exists(png_name):
+        self.draw_frame(frame_num, png_name)
+
+    return png_name
+
+  def draw_frame(self, frame_num, png_name):
+    update_input(self.controller, update=False)
+
+    renderers = self.hidden_window.GetRenderers()
+    renderers.InitTraversal()
+    ren = renderers.GetNextItem()
+    i=0
+    while ren:
+      i+=1
+      actors = ren.GetActors()
+      actors.InitTraversal()
+      actor = actors.GetNextItem()
+      j=0
+      while actor:
+        j+=1
+        m = actor.GetMapper()
+        m.Update()
```

Comment:
I just copied the `update_input` function's `update` section from [VTKPlots.py](https://github.com/UV-CDAT/uvcdat/blob/6a25e00622c953ecc4dd8ffdd8f177f0daf3b350/Packages/vcs/Lib/VTKPlots.py#L1845-L1863) and applied it to my background render window. I'm more than happy to swap out for another way of doing this, I was just trying to do a minimum of inventing new logic.